app/js/shortcuts.js

```diff
@@ -0,0 +1,188 @@
+            key: shortcut,
+            failed : function(err) {
+              console.error(err);
+            }
+          };
+          curShortcuts[option.key] = new nw.Shortcut(option);
+          nw.App.registerGlobalHotKey(curShortcuts[option.key]);
+        });
+      });
+
+      activeGroups.push(groupName);
+      console.info(`Added ${groupName} shortcuts`);
+    }
+  }
+
+  /**
+   * Choose a group of shortcuts to remove.
+   *
+   * @param {String} groupName Which group of shortcuts should be removed.
+   *                           eg main or confirm.
+   */
+  function removeShortcuts(groupName) {
+    // Check the shortcut group can be removed.
+    if (activeGroups.includes(groupName)) {
+      // Iterate through each feature of the shortcut group
+      Object.keys(allShortcuts[groupName]).forEach(function(featureName) {
+        // Iterate through each feature's array of shortcuts
+        allShortcuts[groupName][featureName]["keys"].forEach(function(shortcut) {
+          nw.App.unregisterGlobalHotKey(curShortcuts[shortcut]);
+        });
+      });
+
+      activeGroups.splice(activeGroups.indexOf(groupName));
+      console.info(`Removed ${groupName} shortcuts`);
+    }
+  }
+
+  /**
+   * Load a JSON file containing keyboard shortcuts.
+   * Then match the shortcuts with their function.
+   *
+   * @param {String} location Location of file containing shortcut list.
+   */
+  function getShortcuts(location) {
+    // Location is a parameter to allow for custom shortcuts in the future.
+    if (location === "default") {
+      location = "./json/default-shortcuts.json";
+    }
+    file.read(location, {
+      success: function(data) {
+        data = JSON.parse(data);
+
+        // Iterate through each feature group object (eg main, confirm)
+        Object.keys(features).forEach(function(groupName) {
+
+          // Create an object for the group
+          allShortcuts[groupName] = {};
+
+          // Iterate through each feature in the group (eg takePicture, undoFrame)
+          Object.keys(features[groupName]).forEach(function(featureName) {
+            // Create an object for the feature
+            allShortcuts[groupName][featureName] = {};
+
+            var featureObject = {
+              // The feature's function
+              active : features[groupName][featureName],
+              // Array of keyboard shortcuts for the feature
+              keys: data[groupName][featureName],
+            };
+            allShortcuts[groupName][featureName] = featureObject;
+          });
+        });
+
+        console.info(`Got shortcuts from "${location}":`);
+        console.log(allShortcuts);
+        addShortcuts("main");
+      }
+    });
+  }
+
+  /**
+   * Pause keyboard shortcut operation.
+   */
+  function pauseShortcuts() {
+    activeGroups.forEach(function(groupName) {
+      removeShortcuts(groupName);
+      pausedGroups.push(groupName);
+    });
+    console.info(`Paused shortcuts`);
```

Comment:
Message does not use interpolation, should be written using double quotes instead.